src/cli.rs

```diff
@@ -0,0 +1,821 @@
+  /// Edit the value of a specific secret
+  Edit(Edit),
+  /// Show the value of a specific secret
+  Show(Show),
+}
+
+/// Retrieve a list of secrets
+#[derive(Args, PartialEq, Debug)]
+pub struct List {}
+
+/// Edit the value of a specific secret
+#[derive(Args, PartialEq, Debug)]
+pub struct Edit {
+  /// ID of the secret to edit
+  pub secret_id: String,
+  #[clap(
+    value_enum,
+    short = 'f',
+    long = "secret-format",
+    default_value = "json"
+  )]
+  /// Format of the secret as stored by the provider
+  pub secret_format: DataFormat,
+  /// Format to edit the secret in
+  #[clap(value_enum, short = 'e', long = "edit-format", default_value = "yaml")]
+  pub edit_format: DataFormat,
+}
+
+/// Show the value of a specific secret
+#[derive(Args, PartialEq, Debug)]
+pub struct Show {
+  /// ID of the secret to edit
+  pub secret_id: String,
+  /// Format to print the secret in
+  #[clap(value_enum, short = 'p', long = "print-format", default_value = "yaml")]
+  pub print_format: DataFormat,
+  #[clap(
+    value_enum,
+    short = 'f',
+    long = "secret-format",
+    default_value = "json"
+  )]
+  /// Format of the secret as stored by the provider
+  pub secret_format: DataFormat,
+}
+
+/// Enum to describe the different data formats that can be used with Secrets
+#[derive(Clone, Debug, Deserialize, Serialize, ValueEnum, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum DataFormat {
+  /// Json format
+  Json,
+  /// Yaml format
+  Yaml,
+  /// Plaintext format
+  Text,
+}
+
+impl std::fmt::Display for DataFormat {
+  fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+    std::fmt::Debug::fmt(self, f)
+  }
+}
+
+impl ArgumentParser {
+  /// Runs the given subcommand and uses the provided client
+  ///
+  /// # Arguments
+  /// * `client` - Trait object that implements [QuerySecrets]
+  ///
+  #[cfg(not(tarpaulin_include))]
+  pub async fn run_subcommand(&self, client: impl QuerySecrets) {
+    let result = match &self.command {
+      Commands::List(args) => {
+        let result = list(client, args).await;
+
+        match result {
+          Ok(list) => println!("{}", list),
+          Err(error) => println!("{}", error),
+        }
+
+        Ok(())
+      }
+      Commands::Edit(args) => edit(client, args).await,
+      Commands::Show(args) => show(client, args).await,
+    };
+
+    if let Err(error) = result {
+      println!("{}", error);
+    }
+  }
+}
+
+async fn list(client: impl QuerySecrets, _args: &List) -> Result<String, NysmError> {
+  let secrets_list = client.secrets_list().await?;
+
+  Ok(secrets_list.table_display())
+}
+
+async fn show(client: impl QuerySecrets, args: &Show) -> Result<(), NysmError> {
+  let secret_value = client.secret_value(args.secret_id.clone()).await?;
+  let secret = secret_value.secret;
+
+  let formatted_secret = reformat_data(&secret, &args.secret_format, &args.print_format)?;
+  let _ = pretty_print(formatted_secret, &args.print_format);
+
+  Ok(())
+}
```

Comment:
This `secret` variable is just pointing to another value, and is only used once. Can't you just put `&secret_value.secret` on line 132, and remove line 130?

Reply:
Mostly for formatting reasons. Doing what you've suggested would push the method over several lines. 

I don't mind either way, though if you think it's better with the suggested change I'll adjust.

Reply:
I don't care about code wrapping over like that, but if you do there's no reason for changing it. 
It gets the point across just fine either way.